x/stake/types/validator.go

```diff
@@ -271,16 +271,16 @@ func NewDescription(moniker, identity, website, details string) Description {
 // UpdateDescription updates the fields of a given description. An error is
 // returned if the resulting description contains an invalid length.
 func (d Description) UpdateDescription(d2 Description) (Description, sdk.Error) {
-	if d.Moniker == doNotModifyDescVal {
+	if d2.Moniker == doNotModifyDescVal {
```

Comment:
I thought the point of this code is that if it is "[do-not-modify]" it can't be altered. 

This now allows the description to be modified even if its currently "[do-not-modify]"

This is presuming that d2 is the updated desc (the variable should be renamed regardless imo, to make the code + godoc clearer).

Reply:
@ValarDragon you got it backwards - the point of [do-not-modify] is exclusively for the flag usage, not supposed to ever be stored in actual descriptions. How Sunny modified this here corrects the code to its original intention